ansible/roles/open-env-azure-add-user-to-subscription/tasks/main.yml

```diff
@@ -0,0 +1,61 @@
+---
+- name: Log into Azure CLI
+  command: >
+    az login --service-principal -u {{ azure_service_principal }}
+    -p {{ azure_password }} --tenant {{ azure_tenant }}
+
+- name: Check if email is Red Hat associate
+  fail:
+    msg: User is not a Red Hat associate
+  when:
+    - '"@redhat.com" not in email'
+
+- name: Checking if user is in Active Directory
+  azure.azcollection.azure_rm_aduser_info:
+    auth_source: cli
+    user_principal_name: "{{ email }}"
+    tenant: "{{ azure_tenant }}"
+  register: azuser
+
+- name: Retrieving an available pool ID and locking it in CosmosDB
+  command: >
+      curl {{ az_function_get }}{{ project_tag }}
+  register: poolid
+
+- name: Write out the assigned pool ID
+  debug:
+    msg: "{{ poolid.stdout }}"
+
+- name: Get facts for assigned subscription by pool ID name
+  azure.azcollection.azure_rm_subscription_info:
+    auth_source: cli
+    name: "{{ poolid.stdout }}"
+  register: assignedsubscription
+
+- name: Write out assignedsubscription
+  debug:
+    msg: "{{ assignedsubscription }}"
+
+- name: Get subscription FQID
+  set_fact:
+    subscription: "{{ assignedsubscription['subscriptions'][0].fqid }}"
+
+- name: Tag the assigned Azure subscription with the users email and guid
+  command: >
+    az tag create --resource-id {{ subscription }} --tags GUID={{ guid }} EMAIL={{ email }}
+
+- name: Get CONTRIBUTOR Role Definition
+  azure.azcollection.azure_rm_roledefinition_info:
+    auth_source: cli
+    scope: "{{ subscription }}"
+    role_name: Contributor
+  register: contributor_role_definition
+
+- name: Set user as CONTRIBUTOR for the subscription
+  azure.azcollection.azure_rm_roleassignment:
+    auth_source: cli
+    scope: "{{ subscription }}"
+    assignee_object_id: "{{ azuser['ad_users'][0].object_id }}"
```

Comment:
Nit pick... odd to switch between styles `azuser['ad_users'][0].object_id` rather than a consistent `azuser.ad_users[0].object_id` or `azuser['ad_users'][0]['object_id']`?